classes/Xml/ChecksumCompare.php

```diff
@@ -80,20 +107,25 @@ public function getFilesDiffBetweenVersions(string $version1, ?string $version2)
      * returns an array of files which are present in PrestaShop version $version and has been modified
      * in the current filesystem.
      *
-     * @return array{'mail':string[],'translation':string[],'core':string[]}|false
+     * @return array{
+     *                'mail':array{'missing':string[],'altered':string[]},
+     *                'translation':array{'missing':string[],'altered':string[]},
+     *                'core':array{'missing':string[],'altered':string[]},
+     *                'themes':array{'missing':string[],'altered':string[]}
+     *                }|false
      */
     public function getTamperedFilesOnShop(string $version)
     {
-        if (is_array($this->changed_files) && count($this->changed_files['core']) == 0) {
+        if (is_array($this->file_differences) && count($this->file_differences['core']['altered']) == 0) {
             $checksum = $this->fileLoader->getXmlMd5File($version);
             if (!$checksum) {
-                $this->changed_files = false;
+                $this->file_differences = false;
             } else {
                 $this->browseXmlAndCompare($checksum->ps_root_dir[0]);
             }
         }
 
-        return $this->changed_files;
+        return $this->file_differences;
     }
```

Comment:
While it is fresh in your mind, I think it a good time to add unit tests on this class.